Can scm (dev) dependencies be installed before running tests?

Hi! I'm using this action in my project, and am working on a new feature - I'm adding more robust LSP integration  (https://github.com/mikavilpas/yazi.nvim/pull/190). To do this, I need to add a dependency to this rock: https://luarocks.org/modules/neorocks/nvim-lsp-file-operations

Looks like that rock specifies an `scm` version - I take this to mean "source control management" aka the latest, bleeding edge version. Installing this seems to result in

> Run luarocks test --local
> Missing dependencies for yazi.nvim scm-1:
>   nvim-lsp-file-operations >= scm (not installed)
>
> yazi.nvim scm-1 depends on lua >= 5.1 (5.1-1 provided by VM: success)
> yazi.nvim scm-1 depends on plenary.nvim (scm-1 installed: success)
> yazi.nvim scm-1 depends on nvim-lsp-file-operations >= scm (not installed)
>
> Error: Could not satisfy dependency nvim-lsp-file-operations >= scm: No results matching query were found for Lua 5.1.
> Error: Process completed with exit code 1.
>
> https://github.com/mikavilpas/yazi.nvim/actions/runs/9925762312/job/27418523508#step:6:157

I'm not too familiar with luarocks, so any help would be appreciated.

--- a/README.md
+++ b/README.md
@@ -80,7 +80,7 @@ busted --lua nlua spec/mytest_spec.lua
 
 ## Inputs
 
-### `nvim_version`
+### `nvim_version` (optional)
 
 Version of Neovim to install. Valid values are `stable`, `nightly` or version tag such
 as `v0.9.2`. 
@@ -92,7 +92,7 @@ as `v0.9.2`.
 > This value must exactly match to a tag name 
 > when installing the specific version.
 
-### `luarocks_version`
+### `luarocks_version` (optional)
 
 Version of LuaRocks[^2] to install.
 
@@ -100,6 +100,12 @@ Version of LuaRocks[^2] to install.
 
 [^2]: This action uses [LuaRocks](https://luarocks.org/)
       to install `busted` and test dependencies.
+
+### `before` (optional)
+
+A bash script to run before running the tests.
+For example, you can use this to install additional luarocks packages
+that can't be installed automatically using `luarocks test`.
 
 ## Resources
 

--- a/action.yml
+++ b/action.yml
@@ -14,9 +14,13 @@ inputs:
     required: false
     default: 'stable'
   luarocks_version:
-    description: 'Version of LuaRocks to install.'
+    description: Version of LuaRocks to install.
     required: false
     default: '3.11.1'
+  before:
+    description: Script to run before running tests.
+    required: false
+    default: ''
 runs:
   using: "composite"
   steps:
@@ -56,6 +60,11 @@ runs:
       # if: steps.cache-luarocks.outputs.cache-hit != 'true'
       with:
         luarocksVersion: ${{ inputs.luarocks_version }}
+
+    - name: Run 'before' script
+      if: ${{ inputs.before != '' }}
+      run: ${{ inputs.before }}
+      shell: bash
 
     - name: Install busted and nlua
       if: steps.cache-luarocks.outputs.cache-hit != 'true'

